net/DevExtreme.AspNet.Data/ExpressionCompiler.cs

```diff
@@ -46,6 +46,13 @@ abstract class ExpressionCompiler {
             if(forceToString && currentTarget.Type != typeof(String))
                 progression.Add(Expression.Call(currentTarget, typeof(Object).GetMethod(nameof(Object.ToString))));
 
+            if(_guardNulls && progression.Count > 1 || liftToNullable && progression.Count > 2) {
```

Comment:
Notes on the condition:

- `_guardNulls` is `true` only for LINQ to Objects. In this mode, we expect `obj` itself can be `null`. We need to prevent NRE and "Nullable object must have a value" errors even for simple accessors like `obj.Member`.

- For providers other than LINQ to Objects (EF, etc), null conditionals are not used, and also we don't expect that database entities can be `null`. Therefore, lift to `Nullable` is only used for longer accessors like `obj.Name.Length` or `obj.Date.Year`.

- `liftToNullable` is `true` only for certain subsclasses of `ExpressionCompiler`.
  - Filter and Sort compilers use `.Where()` and `.OrderBy()`/`.ThenBy()`. They  don't materialize the results of their selectors. So "cast to value type failed because the materialized value is null" errors cannot happen.
  - Select and remote group compilers use `.Select()` for which conversion to `Nullable` is needed.